src/app/api/v1/admin/removals/route.ts

```
@@ -0,0 +1,91 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { OnerepScanResultRow } from "knex/types/tables";
+import { NextRequest, NextResponse } from "next/server";
+import {
+  getStuckRemovals,
+  refreshStuckRemovals,
+} from "../../../../functions/server/getStuckRemovals";
+import { getServerSession } from "../../../../functions/server/getServerSession";
+import { isAdmin } from "../../../utils/auth";
+
+export async function GET(req: NextRequest) {
+  const session = await getServerSession();
+  const searchParams = req.nextUrl.searchParams;
+
+  if (isAdmin(session?.user?.email || "")) {
+    if (!searchParams.has("days")) {
+      return NextResponse.json({
+        success: false,
+        status: 422,
+        message: "days parameter is required",
+      });
+    }
+
+    const days = parseInt(searchParams.get("days") ?? "30");
+
+    if (searchParams.get("refresh") === "true") {
+      await refreshStuckRemovals(days);
```

without try-catch blocks, we might swallow some errors / unhandled exceptions